devices/Bmm150/Bmm150.cs

```diff
@@ -0,0 +1,359 @@
+            ReadBytes(Register.BMM150_DIG_Z4_LSB, trim_xyz_data);
+            ReadBytes(Register.BMM150_DIG_Z2_LSB, trim_xy1xy2_data);
+
+            return new Bmm150TrimRegisterData(trim_x1y1_data, trim_xyz_data, trim_xy1xy2_data);
+        }
+
+        /// <summary>
+        /// Starts the Bmm150 init sequence
+        /// </summary>
+        private void Initialize()
+        {
+            // Set Sleep mode
+            WriteRegister(Register.POWER_CONTROL_ADDR, 0x01);
+            Wait(6);
+
+            // Check for a valid chip ID
+            if (!IsVersionCorrect())
+            {
+                throw new IOException($"This device does not contain the correct signature (0x32) for a Bmm150");
+            }
+
+            // Set operation mode to: "Normal Mode"
+            WriteRegister(Register.OP_MODE_ADDR, 0x00);
+        }
+
+        /// <summary>
+        /// Get the device information
+        /// </summary>
+        /// <returns>The device information</returns>
+        public byte GetDeviceInfo() => ReadByte(Register.INFO);
+
+        /// <summary>
+        /// Calibrate the magnetometer. 
+        /// Please make sure you are not close to any magnetic field like magnet or phone
+        /// Please make sure you are moving the magnetometer all over space, rotating it.
+        /// </summary>
+        /// <param name="numberOfMeasurements">Number of measurement for the calibration, default is 100</param>
+        // https://platformio.org/lib/show/12697/M5_BMM150
+        public void CalibrateMagnetometer(int numberOfMeasurements = 100)
+        {
+            Vector3 mag_min = new Vector3() { X = 9000, Y = 9000, Z = 30000 };
+            Vector3 mag_max = new Vector3() { X = -9000, Y = -9000, Z = -30000 };
+            Vector3 rawMagnetometerData;
+
+            for (int i = 0; i < numberOfMeasurements; i++)
+            {
+                try
+                {
+                    rawMagnetometerData = ReadMagnetometerWithoutCorrection();
+
+                    if (rawMagnetometerData.X != 0)
+                    {
+                        mag_min.X = (rawMagnetometerData.X < mag_min.X) ? rawMagnetometerData.X : mag_min.X;
+                        mag_max.X = (rawMagnetometerData.X > mag_max.X) ? rawMagnetometerData.X : mag_max.X;
+                    }
+
+                    if (rawMagnetometerData.Y != 0)
+                    {
+                        mag_max.Y = (rawMagnetometerData.Y > mag_max.Y) ? rawMagnetometerData.Y : mag_max.Y;
+                        mag_min.Y = (rawMagnetometerData.Y < mag_min.Y) ? rawMagnetometerData.Y : mag_min.Y;
+                    }
+
+                    if (rawMagnetometerData.Z != 0)
+                    {
+                        mag_min.Z = (rawMagnetometerData.Z < mag_min.Z) ? rawMagnetometerData.Z : mag_min.Z;
+                        mag_max.Z = (rawMagnetometerData.Z > mag_max.Z) ? rawMagnetometerData.Z : mag_max.Z;
+                    }
+
+                    Wait(100);
```

Comment:
Is this 100 the correct number? a simple comment would help to understand

Reply:
Not sure here, it just seemed plausible to wait for 100ms, this means we could top out 10 reads/sec * 100 iterations = 1000 sensor reads.